### PR TITLE
Extension utils

### PR DESCRIPTION
**Pull request by Wardergrip**
These are a bunch of commonly used extension methods:
- Extensions with distance of objects
- LockAxis enum
- Random value of a list or array
- Spawn inactive
- Lerp positions/transforms
- QOL child functions

### CODE REVIEW

File: ThatDefaultProject/Assets/ThatDefaultProject/Scripts/LerpExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+			int index1 = Mathf.FloorToInt(t * (path.Length - 1));
+			int index2 = Mathf.CeilToInt(t * (path.Length - 1));
+
+			index1 = Mathf.Clamp(index1, 0, path.Length - 1);
+			index2 = Mathf.Clamp(index2, 0, path.Length - 1);
+
+			Vector3 point1 = path[index1].position;
+			Vector3 point2 = path[index2].position;
+			return Vector3.Lerp(point1, point2, t - index1);
+		}
```
Comment (AtlantiaKing) [inline ThatDefaultProject/Assets/ThatDefaultProject/Scripts/LerpExtensions.cs:286]: I think LerpQuaternions should be here as well

File: ThatDefaultProject/Assets/ThatDefaultProject/Scripts/MonobehaviorExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+				return comp;
+			}
+			comp = mb.GetComponent<T>();
+			if (comp != null)
+			{
+				return comp;
+			}
+			return mb.gameObject.AddComponent<T>();
+		}
+		public static void SetActiveAll<T>(this IEnumerable<T> container, bool state) where T : Component
```
Comment (AtlantiaKing) [inline ThatDefaultProject/Assets/ThatDefaultProject/Scripts/MonobehaviorExtensions.cs:36]: I would rename this to "SetEnabledAll" so match the syntax Unity uses for Components comp.enabled = true/false

File: ThatDefaultProject/Assets/ThatDefaultProject/Scripts/SpawningUtils.cs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+			(GameObject coreObject, Transform coreObjectTransform) = CreateDisabledCoreObject(parent);
+			T instance = Object.Instantiate(original, position, rotation, coreObjectTransform);
+			SetActiveState(instance, false);
+			SetParent(instance, parent, false);
+			Object.Destroy(coreObject);
+			return instance;
+		}
+
+		private static (GameObject coreObject, Transform coreObjectTransform) CreateDisabledCoreObject(Transform parent = null)
```
Comment (AtlantiaKing) [inline ThatDefaultProject/Assets/ThatDefaultProject/Scripts/SpawningUtils.cs]: Wouldn't you also call this Instantiate? so InstantiateDisabledCoreObject
I also think RootObject is a better name, but that's just personal taste

File: ThatDefaultProject/Assets/ThatDefaultProject/Scripts/LerpExtensions.cs (hunk 1 above)
Comment (Wardergrip) [reply ThatDefaultProject/Assets/ThatDefaultProject/Scripts/LerpExtensions.cs:286]: Resolved with commit 3d3a171e473e0cc88d49b8524ab9817afe1532e2

File: ThatDefaultProject/Assets/ThatDefaultProject/Scripts/MonobehaviorExtensions.cs (hunk 2 above)
Comment (Wardergrip) [reply ThatDefaultProject/Assets/ThatDefaultProject/Scripts/MonobehaviorExtensions.cs:36]: Added 2 same functions that do the exact same but are named `SetEnableAll` in 7d6c2c26529ddbeb3f6a7e12edcf5a97b303d02f

File: ThatDefaultProject/Assets/ThatDefaultProject/Scripts/SpawningUtils.cs (hunk 3 above)
Comment (Wardergrip) [reply ThatDefaultProject/Assets/ThatDefaultProject/Scripts/SpawningUtils.cs]: Fixed in ec3151440d9f560d2bbfa68b2195ee64841c0ccb

**Review by AtlantiaKing — APPROVED** [review]
Great additions, everything looks fine.

File: ThatDefaultProject/Assets/ThatDefaultProject/Scripts/MonobehaviorExtensions.cs (hunk 2 above)
Comment (AtlantiaKing) [reply ThatDefaultProject/Assets/ThatDefaultProject/Scripts/MonobehaviorExtensions.cs:36]: That works!

**Pull request merged**